servers/src/main/java/tachyon/master/lineage/LineageMaster.java

```diff
@@ -0,0 +1,415 @@
+                  mTachyonConf.getInt(Constants.MASTER_LINEAGE_RECOMPUTE_INTERVAL_MS)));
+    }
+  }
+
+  @Override
+  public void stop() throws IOException {
+    super.stop();
+    if (mCheckpointExecutionService != null) {
+      mCheckpointExecutionService.cancel(true);
+    }
+    if (mRecomputeExecutionService != null) {
+      mRecomputeExecutionService.cancel(true);
+    }
+  }
+
+  @Override
+  public void streamToJournalCheckpoint(JournalOutputStream outputStream) throws IOException {
+    mLineageStore.streamToJournalCheckpoint(outputStream);
+  }
+
+  public LineageStoreView getLineageStoreView() {
+    return new LineageStoreView(mLineageStore);
+  }
+
+  public long createLineage(List<TachyonURI> inputFiles, List<TachyonURI> outputFiles, Job job)
+      throws InvalidPathException, FileAlreadyExistException, BlockInfoException {
+    // validate input files exist
+    List<TachyonFile> inputTachyonFiles = Lists.newArrayList();
+    for (TachyonURI inputFile : inputFiles) {
+      long fileId;
+      fileId = mFileSystemMaster.getFileId(inputFile);
+      inputTachyonFiles.add(new TachyonFile(fileId));
+    }
+    // create output files
+    List<LineageFile> outputTachyonFiles = Lists.newArrayList();
+    for (TachyonURI outputFile : outputFiles) {
+      long fileId;
+      fileId = mFileSystemMaster.createFile(outputFile, 1, true);
+      outputTachyonFiles.add(new LineageFile(fileId));
+    }
+
+    LOG.info("Create lineage of input:" + inputTachyonFiles + ", output:" + outputTachyonFiles
+        + ", job:" + job);
+    long lineageId = mLineageStore.createLineage(inputTachyonFiles, outputTachyonFiles, job);
+
+    writeJournalEntry(mLineageIdGenerator.toJournalEntry());
+    writeJournalEntry(mLineageStore.getLineage(lineageId).toJournalEntry());
+    flushJournal();
+    return lineageId;
+  }
+
+  public boolean deleteLineage(long lineageId, boolean cascade)
+      throws LineageDoesNotExistException, LineageDeletionException {
+    Lineage lineage = mLineageStore.getLineage(lineageId);
+    if (lineage == null) {
+      throw new LineageDoesNotExistException(
+          "the lineage " + lineageId + " to delete does not exist");
+    }
+
+    // there should not be child lineage if cascade
+    if (!cascade && !mLineageStore.getChildren(lineage).isEmpty()) {
+      throw new LineageDeletionException(
+          "the lineage " + lineageId + " to delete has children lineages");
+    }
+
+    LOG.info("Delete lineage " + lineageId);
+    mLineageStore.deleteLineage(lineageId);
+    return true;
+  }
+
+  /**
+   * Reinitializes the file when the file is lost or not completed.
+   *
+   * @param path the path to the file
+   * @param blockSizeBytes the block size
+   * @param ttl the TTL
+   * @return the id of the reinitialized file when the file is lost or not completed, -1 otherwise.
+   * @throws InvalidPathException the file path is invalid
+   */
+  public long recreateFile(String path, long blockSizeBytes, long ttl)
```

Comment:
What is the importance of this method? Why can a user set the block size and ttl?

Should this take createOptions or the identical arguments as createFile()?